lib/ui/home/home_screen.dart

```diff
@@ -1,29 +1,176 @@
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:smooth_page_indicator/smooth_page_indicator.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
+import 'package:survey_flutter_ic/ui/home/home_footer_widget.dart';
+import 'package:survey_flutter_ic/ui/home/home_header_widget.dart';
+import 'package:survey_flutter_ic/ui/home/home_state.dart';
+import 'package:survey_flutter_ic/ui/home/home_view_model.dart';
+import 'package:survey_flutter_ic/utils/dimension.dart';
 
-class HomeScreen extends StatelessWidget {
+import '../../di/di.dart';
+import '../../gen/assets.gen.dart';
+import '../../usecases/get_surveys_use_case.dart';
+import 'home_side_menu.dart';
+import 'home_side_menu_ui_model.dart';
+import 'loading/home_loading.dart';
+
+final homeViewModelProvider =
+    StateNotifierProvider.autoDispose<HomeViewModel, HomeState>(
+  (_) => HomeViewModel(
+    getSurveysUseCase: getIt.get<GetSurveysUseCase>(),
+  ),
+);
+
+class HomeScreen extends ConsumerStatefulWidget {
   const HomeScreen({super.key});
 
   @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('Home Screen'),
-      ),
-      endDrawer: SideMenu(
+  HomeScreenState createState() => HomeScreenState();
+}
+
+class HomeScreenState extends ConsumerState<HomeScreen> {
+  final _pageController = PageController();
+  final _currentPageIndex = ValueNotifier<int>(0);
+
+  SideMenu get _sideMenu => SideMenu(
         sideMenuUIModel: SideMenuUIModel(
           name: 'Mai',
           version: 'v0.1.0 (1562903885)',
         ),
-      ),
-      backgroundColor: Colors.black,
+      );
+
+  Widget _imageBackground(SurveyModel surveyModel) => Container(
+        decoration: BoxDecoration(
+          image: DecorationImage(
+            image: FadeInImage.assetNetwork(
+              placeholder: Assets.images.nimbleLogo.path,
+              image: surveyModel.coverImageUrl,
+            ).image,
+            fit: BoxFit.cover,
+            colorFilter: ColorFilter.mode(
+              Colors.black.withOpacity(0.35),
+              BlendMode.darken,
+            ),
+          ),
+        ),
+      );
+
+  Widget get _emptyBackground => Image(
+        image: AssetImage(Assets.images.bgLoginOverlay.path),
+        fit: BoxFit.fill,
+      );
+
+  List<Widget> _homeBackground(List<SurveyModel>? surveys) {
+    if (surveys == null || surveys.isEmpty) {
+      return [_emptyBackground];
+    } else {
+      return surveys.map((item) => _imageBackground(item)).toList();
+    }
+  }
+
+  Widget _homeHeaderWidget({
+    required BuildContext context,
+    required VoidCallback openSideMenuCallback,
+  }) {
+    return HomeHeaderWidget(
+      profileImgUrl:
+          'https://secure.gravatar.com/avatar/6733d09432e89459dba795de8312ac2d',
+      profileImgClickCallback: openSideMenuCallback,
+    );
+  }
```

Comment:
As we adopt  `ConsumerState` for `HomeScreenState`, so no need to define `context` as parameter, as `context` now is an instance variable of `HomeScreenState`.

Also I don't think we need to define `openSideMenuCallback` here. Just simply call the function:

```suggestion
  Widget _homeHeaderWidget() => HomeHeaderWidget(
      profileImgUrl: 'https://secure.gravatar.com/avatar/6733d09432e89459dba795de8312ac2d',
      profileImgClickCallback: () => scaffoldKey.currentState?.openEndDrawer(),
    );
```